doc/matter/index.rst

```diff
@@ -1,7 +1,7 @@
 .. _matter_index:
 
-Matter fork documentation
-#########################
+Matter documentation
+####################
 
```

Comment:
Can we keep "fork" in there? I wanted to break the pattern on purpose to highlight the different source.

Reply:
This PR is actually introduced to have a better way to identify forks. Also, the word "fork" in the main title isn't great.

Reply:
OK, I see the point. Would it then be possible to have the subtitle for Matter as "nRF Connect in Matter SDK"? Unlike Zephyr, where we just deploy the docs from sdk-zephyr, so the subtitle "nRF Connect SDK" is clear, Matter has both the fork documentation and the documentation in sdk-nrf, so the subtitle "nRF Connect SDK" can be misleading.

Reply:
is "Matter SDK" our fork of matter?

Reply:
Yes, although to be exact, Matter SDK is the source from which we are forking.

Reply:
Discussed further on Teams: to stay consistent on indicating that certain docsets come from external sources and are only an NCS flavor of the official docs published elsewhere, I'm fine with having nRF Connect SDK as the subtitle for the Matter docset.